lib/analytics.js

```diff
@@ -678,22 +694,32 @@ Analytics.prototype._invoke = function(method, facade) {
 
   var failedInitializations = self.failedInitializations || [];
   each(function(integration, name) {
-    if (!facade.enabled(name)) return;
+    var facadeCopy = extend(true, {}, facade);
+
+    if (!facadeCopy.enabled(name)) return;
     // Check if an integration failed to initialize.
     // If so, do not process the message as the integration is in an unstable state.
     if (failedInitializations.indexOf(name) >= 0) {
       self.log(
-        'Skipping invokation of .%s method of %s integration. Integation failed to initialize properly.',
+        'Skipping invocation of .%s method of %s integration. Integration failed to initialize properly.',
         method,
         name
       );
     } else {
       try {
-        metrics.increment('analytics_js.integration.invoke', {
-          method: method,
-          integration_name: integration.name
+        // Apply any integration middlewares that exist, then invoke the integration with the result.
+        self.IntegrationMiddlewares.applyMiddlewares(facadeCopy.obj, function(
+          payload
+        ) {
+          if (payload === null) return;
```

Comment:
I think this allows for two patterns for skipping payloads from proceeding across the chain:

1. Don't call the next link in the chain
2. Call the next link in the chain with a null payload

I'm not sure we want both. It would be good to encourage one pattern if we can. 

If we allow #2, I think we should adding some debug logs to call out the payload was null.

Reply:
Interesting - I feel like #2 is probably the best direction. That way users can react to null payloads appropriately if need be, instead of locking us into a "null means a full stop" meaning - especially as different middlewares from different authors interact. Does this sound good to you? If so, I'll stick to one pattern (drop the non-null payload requirement) and add those debug logs.

Reply:
Talked over Slack, we're going to match iOS and Android middleware behaviors to keep consistency!